handlers/otel_handler.go

```diff
@@ -0,0 +1,247 @@
+		span.SetAttributes(attribute.String(key, val))
+	}
+}
+
+// getEventStartEndTimestamps sets time-related fields in the emitted telemetry
+// about the request/response cycle.
+//
+// It only sets timestamps if they are present in the captured event, and only
+// computes and includes durations for which there are correct timestamps to based them upon.
+func (handler *otelHandler) getEventStartEndTimestamps(event assemblers.Event) (time.Time, time.Time, []attribute.KeyValue) {
+	var startTime, endTime time.Time
+	attrs := []attribute.KeyValue{
+		attribute.String("meta.event_handled_at", time.Now().String()),
+	}
+
+	switch {
+	case event.RequestTimestamp().IsZero() && event.ResponseTimestamp().IsZero(): // no request or response
+		attrs = append(attrs, attribute.String("meta.timestamps_missing", "request, response"))
+		startTime = time.Now()
+		endTime = startTime
+
+	case event.RequestTimestamp().IsZero(): // have response, no request
+		attrs = append(attrs, attribute.String("meta.timestamps_missing", "request"))
+		startTime = event.ResponseTimestamp()
+		endTime = startTime
+		attrs = append(attrs, attribute.String("http.response.timestamp", event.ResponseTimestamp().String()))
+		attrs = append(attrs, attribute.Int64("meta.response.capture_to_handle.latency_ms", time.Since(event.ResponseTimestamp()).Milliseconds()))
+
+	case event.ResponseTimestamp().IsZero(): // have request, no response
+		attrs = append(attrs, attribute.String("meta.timestamps_missing", "response"))
+		startTime = event.RequestTimestamp()
+		endTime = startTime
+		attrs = append(attrs, attribute.String("http.request.timestamp", event.RequestTimestamp().String()))
+		attrs = append(attrs, attribute.Int64("meta.request.capture_to_handle.latency_ms", time.Since(event.RequestTimestamp()).Milliseconds()))
+
+	default: // the happiest of paths, we have both request and response
+		startTime = event.RequestTimestamp()
+		endTime = event.ResponseTimestamp()
+		attrs = append(attrs, attribute.String("http.request.timestamp", event.RequestTimestamp().String()))
+		attrs = append(attrs, attribute.String("http.response.timestamp", event.ResponseTimestamp().String()))
+		attrs = append(attrs, attribute.Int64("meta.request.capture_to_handle.latency_ms", time.Since(event.RequestTimestamp()).Milliseconds()))
+		attrs = append(attrs, attribute.Int64("meta.response.capture_to_handle.latency_ms", time.Since(event.ResponseTimestamp()).Milliseconds()))
+		attrs = append(attrs, attribute.Int64("duration_ms", event.ResponseTimestamp().Sub(event.RequestTimestamp()).Milliseconds()))
+
+	}
+	return startTime, endTime, attrs
+}
+
+// headerToAttributes converts a http.Header into a slice of OpenTelemetry attributes
+func headerToAttributes(isRequest bool, header http.Header) []attribute.KeyValue {
+	var prefix string
+	if isRequest {
+		prefix = "http.request.header"
+	} else {
+		prefix = "http.response.header"
+	}
+	attrs := []attribute.KeyValue{}
+	for key, val := range header {
```

Comment:
I was working on a test for this and ended up iterating through one more time to get clean strings for header values. I'm not sure if that's right but it did show up more cleanly as attribute values.

Also I adjusted to further match the recommendation of the semantic conventions, which is to lowercase the key and replace hyphens with underscores. 

What do you think? https://github.com/honeycombio/honeycomb-network-agent/commit/8537b56a8abca48f06c9d1d813619acdb5fcb7c4